### PR TITLE
feat: [SSCA-693]: Execution changes for provenance gcr step

### PR DESCRIPTION
**Pull request by inderschera**
## Describe your changes
Initialization Step Changes
- Added a new plugin config for provenance GCR step.
- For starting different plugin, based on provenance step source type, added a switch case to change CIStepInfoType based on provenance step source type.
- Added env variables to be passed to plugin.
- Added separate connector secret env map for provenance step.

NOTE: Since plugin has not been developed yet, provenance will not be generated for GCR step until explicitly added to `io.harness.ssca.execution.provenance.ProvenanceStepGenerator#allowedTypesForProvenance`

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53272)

### CODE REVIEW

**Comment by inderschera** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/CIStepInfoUtils.java [hunk 1]
```diff
@@ -200,14 +202,31 @@ private static StepImageConfig getSecurityStepImageConfig(PluginCompatibleStep s
...
       return getSecurityStepImageConfig(step, ciExecutionConfigService, defaultImageConfig);
     }
     return defaultImageConfig;
   }
 
+  private static CIStepInfoType getStepInfoType(PluginCompatibleStep step) {
+    CIStepInfoType stepInfoType = step.getNonYamlInfo().getStepInfoType();
+    if (stepInfoType == CIStepInfoType.PROVENANCE) {
+      ProvenanceStepInfo stepInfo = (ProvenanceStepInfo) step;
+      switch (stepInfo.getSource().getType()) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/CIStepInfoUtils.java:217]: add null check for getsource

File: 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java [hunk 2]
```diff
@@ -124,16 +163,17 @@ public Map<String, String> getProvenanceStepEnvVariablesAtRuntime(
...
-      envMap.put(PLUGIN_DIGESTS, listToStringSlice(digests));
-      envMap.put(PLUGIN_TAGS, listToStringSlice(tags));
-      envMap.put(PROVENANCE_PREDICATE, JsonUtils.asJson(provenanceArtifact.getPredicate()));
+      populateDockerEnvVariables(provenanceStepInfo, identifier, envMap);
+    }
+    if (provenanceStepInfo.getSource() != null
+        && provenanceStepInfo.getSource().getType() == ProvenanceSourceType.GCR) {
+      populateGcrEnvVariables(provenanceStepInfo, identifier, envMap);
     }
+    List<String> digests = imageArtifacts.stream().map(PublishedImageArtifact::getDigest).collect(Collectors.toList());
```
Comment (harshjain12) [inline 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java:172]: add null check

File: 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java [hunk 3]
```diff
@@ -124,16 +163,17 @@ public Map<String, String> getProvenanceStepEnvVariablesAtRuntime(
...
+    List<String> tags = imageArtifacts.stream().map(PublishedImageArtifact::getTag).collect(Collectors.toList());
```
Comment (harshjain12) [inline 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java:173]: add null check

File: 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java (hunk 2 above)
Comment (harshjain12) [reply 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java:172]: Add isEmpty on line 156

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/CIStepInfoUtils.java (hunk 1 above)
Comment (inderschera) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/CIStepInfoUtils.java:220]: done

File: 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java (hunk 2 above)
Comment (inderschera) [reply 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java:178]: done

File: 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java (hunk 3 above)
Comment (inderschera) [reply 332-ssca/ci/src/main/java/io/harness/ssca/execution/ProvenancePluginHelper.java:179]: done

**Comment by inderschera** [comment]
trigger uts

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**